build/linux/Makefile.in

```diff
@@ -51,73 +61,60 @@ LIBS      += -lcudart -lcufft
 NVCC       = @NVCC@
 endif
 
-ifeq ($(octave),yes)
-OCTLDFLAGS:=$(LDFLAGS)
-CPPFLAGS += @OCTAVE_CPPFLAGS@
-ifeq ($(cuda),yes)
-OCTFLAGS=-DASTRA_CUDA
-else
-OCTFLAGS=
-endif
-endif
-
 ifeq ($(matlab),yes)
+# TODO: Do we also want -fopenmp for octave?
 CPPFLAGS+=-I$(MATLAB_ROOT)/extern/include -DMATLAB_MEX_FILE
 CXXFLAGS+=-fopenmp
 LDFLAGS+=-fopenmp
 endif
 
+# MODLDFLAGS are the base LDFLAGS for matlab, octave, python modules
+MODLDFLAGS=$(LDFLAGS) -L$(abs_top_builddir)/.libs
+ifeq ($(install_type),module)
+MODLDFLAGS+=-Wl,-rpath,'\$$ORIGIN'
+endif
+
 ifeq ($(python),yes)
 PYTHON      = @PYTHON@
-PYLIBDIR = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_var; import six; six.print_(get_config_var("LIBDIR"))')
-PYINCDIR = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_python_inc; import six; six.print_(get_python_inc())')
+PYLIBDIR := $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_var; import six; six.print_(get_config_var("LIBDIR"))')
```

Comment:
In principle yes, but at such an early stage I think it's better to rely on functionality from the standard library rather than an external module. After all, the dependency on `six` may not be carved in stone.